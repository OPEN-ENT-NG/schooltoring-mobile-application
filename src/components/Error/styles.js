import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  image: {
    width: 200,
    height: 217,
    position: "absolute",
    bottom: 0,
    left: "10%"
  }
});

export default styles;
