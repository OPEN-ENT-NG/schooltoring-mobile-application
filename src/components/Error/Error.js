import React from "react";
import { View, Text, Image } from "react-native";
import PropTypes from "prop-types";

import styles from "./styles";

const Error = props => {
  return (
    <View style={styles.container}>
      {typeof props.message === "string" ? (
        <Text>{props.message}</Text>
      ) : (
        props.message
      )}
      <Image
        source={require("../../assets/img/gars1b.png")}
        style={styles.image}
      />
    </View>
  );
};

Error.propTypes = {
  message: PropTypes.any.isRequired
};

export default Error;
