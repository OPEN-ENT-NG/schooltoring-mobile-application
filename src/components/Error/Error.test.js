import React from "react";
import { Text } from "react-native";

import { shallow } from "enzyme";

import Error from "./Error";

const message = "Error message";
const messageComponent = (
  <Text id="error-message-component">Error message 2</Text>
);

describe("Render", () => {
  test("Error page should render without crashing", () => {
    const rendered = shallow(<Error message={message} />);

    expect(rendered).toMatchSnapshot();
  });

  test("Error page should render a simple text", () => {
    const rendered = shallow(<Error message={message} />);

    expect(
      rendered
        .children()
        .at(0)
        .contains(message)
    );
  });

  test("Error page should render a React Component", () => {
    const rendered = shallow(<Error message={messageComponent} />);

    expect(
      rendered.find("[id='error-message-component']").exists()
    ).toBeTruthy();
  });
});
