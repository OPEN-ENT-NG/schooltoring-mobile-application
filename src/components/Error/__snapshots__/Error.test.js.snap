// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Error page should render without crashing 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Error message
  </Text>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": 217,
        "left": "10%",
        "position": "absolute",
        "width": 200,
      }
    }
  />
</Component>
`;
